### ISSUE TITLE
Add support for sidekiq to communicate over HTTPS

### ISSUE THREAD

**Issue by adarnimrod** [bitnami/bitnami-docker-discourse#117]
I request to add support to have sidekiq communicate with the Discourse web server over HTTPS. In our setup, we want to have the 2 containers (sidekiq and the web server) deployed separately. We're currently unable to use a service discovery mechanism, so instead we want to have sidekiq communicate through the load-balancer, but that only exposes HTTPS. We need an option to set the protocol to HTTPS (instead of HTTP). There are similar options already (the `DISCOURSE_HOST` and `DISCOURSE_PORT` environment variables), so this should complement those.

**Comment by andresbono** [comment]
Hi @adarnimrod, we will investigate that option and we will try to implement it. We will keep this issue open in the meantime.

**Comment by prbsparx** [comment]
Hey Andres,

I think I'm seeing an issue that might be related to this feature request.
My sidekiq container is getting an error:
```
Failed to download upload from http://${DISCOURSE_HOSTNAME}/assets/favicon-24b22a63aae0cf353d72b6b728504377bfc42edc6563f202e3a1401763497249.ico for favicon. Retrying...
Error encountered when trying to download file for favicon.
OpenSSL::SSL::SSLError: SSL_connect returned=1 errno=0 state=error: certificate verify failed (self signed certificate in certificate chain)
```

If I set DISCOURSE_HOSTNAME in the sidekiq container to just the k8s discourse service hostname `discourse` sidekiq appears to work...
Just another piece of info.

In one of the docker images/Helm charts I've built I figured out a way to allow a user to pass a root chain. The root chain is provided as a secret, the dev then puts the name of the secret in the  values.yaml and the secret file is then mounted into the OS, and some files modified slightly based off that.

**Comment by andresbono** [comment]
Thanks for sharing this information with us, @prbsparx.

Just to let you know, this feature is still pending to be implemented. Once we release a new version of the container, we will notify on this issue so you can give it a try!

**Comment by marcosbc** [comment]
Hi, FYI we just released a new container version, [2.7.0-debian-10-r4](https://github.com/bitnami/bitnami-docker-discourse#270-debian-10-r4). The initialization logic has received a major refactor, and its scripts are now located in the [`rootfs/` directory](https://github.com/bitnami/bitnami-docker-discourse/blob/master/2/debian-10/rootfs/).

The new images should include a fix for these issues by using the following environment variables in the Sidekiq container:

- `DISCOURSE_ENABLE_HTTPS`: Whether to use HTTPS by default. Default: **no**
- `DISCOURSE_EXTERNAL_HTTP_PORT_NUMBER`: Port to used by WordPress to generate URLs and links when accessing using HTTP. Will be ignored if multisite mode is not enabled. Default **80**
- `DISCOURSE_EXTERNAL_HTTPS_PORT_NUMBER`: Port to used by WordPress to generate URLs and links when accessing using HTTPS. Will be ignored if multisite mode is not enabled. Default **443**

**Comment by adarnimrod** [comment]
Hi. It's been a while since I opened this issue. I no longer work at the same company and I can't test the new release. I'm closing this issue since I think it's been addressed.
